_docs/example_plugin/some_module.py

```diff
@@ -0,0 +1,73 @@
+# python_name: example_plugin._data:fractal
+
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from magicgui import magic_factory
+from qtpy.QtWidgets import QWidget
+
+from npe2.types import LayerData, PathOrPaths, ReaderFunction
+
+if TYPE_CHECKING:
+    import napari.types
+    import napari.viewer
+
+
+def write_points(path: str, layer_data: Any, attributes: Dict[str, Any]) -> List[str]:
+    with open(path, "w") as fh:
+        ...  # save layer_data and attributes to file
+
+    # return path to any file(s) that were successfully written
+    return [path]
+
+
+def get_reader(path: PathOrPaths) -> Optional[ReaderFunction]:
+    # If we recognize the format, we return the actual reader function
+    if isinstance(path, str) and path.endswith("*.xyz"):
+        return xyz_file_reader
+    # otherwise we return None.
+    return None
+
+
+def xyz_file_reader(path: PathOrPaths) -> List[LayerData]:
+    data = ...  # somehow read data from path
+    layer_attributes = {"name": "etc..."}
+    return [(data, layer_attributes)]
+
+
+class AnimationWizard(QWidget):
```

Comment:
Maybe use `ThresholdWizard` to be consistent with other widget examples.

Reply:
I just used `MyWidget`, I don't want to have to reimplement the threshold logic (i.e.finding a layer from the viewer, etc...), so it might be confusing.  Just gonna leave it vague (but change from animation)